### PR TITLE
Convert some `staticmethod` to `classmethod`

### PR DESCRIPTION
**Pull request by DanielYang59**
## Summary

Major changes:

- Converted some `staticmethod` to `classmethod`, partially address #3708

Rationale:

These methods create an instance of the Class itself, and should be a `classmethod`, not `staticmethod`.

## Summary by CodeRabbit

- **Refactor**
  - Enhanced class method flexibility across various modules by converting static methods to class methods and updating method signatures for improved inheritance and extensibility.
  - Introduced type hints for return types to ensure type safety and clarity in method definitions.
  - Streamlined code by leveraging class inheritance, making the codebase more extensible and maintainable.
- **Documentation**
  - Updated method signatures in documentation to reflect changes from static to class methods and the introduction of type hints.

### CODE REVIEW

File: pymatgen/transformations/advanced_transformations.py [hunk 1]
```diff
@@ -40,15 +40,16 @@
...
-    from typing import Any
-
 try:
     import hiphive
 except ImportError:
     hiphive = None
 
+if TYPE_CHECKING:
+    from collections.abc import Iterable, Sequence
+    from typing import Any
```
Comment (DanielYang59) [inline pymatgen/transformations/advanced_transformations.py:50]: I hope I'm understanding this correctly,

```python
try:
    import hiphive
except ImportError:
    hiphive = None
```
Still is the "standard" import block.

While:
```python
if TYPE_CHECKING:
    from collections.abc import Iterable, Sequence
    from typing import Any
```
is the "conditional" import block specific to `TYPE_CHECKING`.

As such the order should be adjusted such that "standard" imports go together.

**Comment by DanielYang59** [comment]
@coderabbitai review

**Comment by DanielYang59** [comment]
@coderabbitai Can you find if there is any other methods decorated as `staticmethod` (returns an instance of the Class itself), but should be a `classmethod` instead, across the entire code base?

**Comment by DanielYang59** [comment]
@janosh, please review. Thanks a lot.

**Review by janosh — APPROVED** [review]
thanks @DanielYang59, great work! really appreciate all your effort bringing `pymatgen` code into shape 👍

**Comment by DanielYang59** [comment]
Glad I could help. Thanks for reviewing @janosh

**Pull request merged**